kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java

```diff
@@ -3913,20 +3919,44 @@ public void onAction(boolean didGrant) {
             }
         });
 
+        imageVideoButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                emoticonsFrameLayout.setVisibility(View.GONE);
+                if (getActivity() != null) {
+                    if (!((KmStoragePermissionListener) getActivity()).isPermissionGranted()) {
+                        //get permission
+                        ((KmStoragePermissionListener) getActivity()).checkPermission(new KmStoragePermission() {
+                            @Override
+                            public void onAction(boolean didGrant) {
+                                //did not get permission
+                                if (!didGrant) {
+                                    return;
+                                }
+                            }
+                        });
+                    } else {
+                        ((ConversationActivity) getActivity()).setFromMultiSelectGallery(true);
+                        ((ConversationActivity) getActivity()).processImageVideo();
+                    }
```

Comment:
Do you think it will be better that this method itself internally set the variable to true

Reply:
We don't need the variable to be set true everytime we all this function.
the variable helps to figure what permission we should look for and doesn't necessarily come under the logic of the processMethods. it is used to check where the flow is coming FROM.
it is better outside.